CIP-0035/README.md

```diff
@@ -99,37 +130,32 @@ Changes to Plutus Core can be released onto Cardano in four ways, with ascending
+- A backwards-incompatible change requires a new Plutus Core ledger language, so that the ledger can distinguish them, and maintain the old behaviour for old scripts.
 
 The following table lists, for each type of change in "Types of change", what kind of release it requires.
 
 | Change                                                                     | Release            | Notes                                                                                               |
 |----------------------------------------------------------------------------|--------------------|-----------------------------------------------------------------------------------------------------|
-| Adding a language construct                                                | HF                 | Backwards-compatible.                                                                               |
-| Removing a language construct                                              | LV                 | This will cause scripts which use this construct to be rejected, so is not backwards compatible.    |
-| Changing the behaviour of a construct in the language                      | LV                 | This changes the behaviour of existing scripts, so is not backwards compatible.                     |
-| Changing the binary format of the language in a backwards-compatible way   | HF                 | Safe even if it makes previously non-deserializable scripts deserializable.[^1]                     |
-| Changing the binary format of the language in a backwards-incompatible way | LV                 |                                                                                                     |
-| Adding a new builtin function or type                                      | HF (rarely LV[^2]) | Backwards-compatible. Requires a binary format change.                                              |
-| Removing a builtin function or type                                        | LV                 | This will cause scripts which use this builtin to be rejected, so is not backwards compatible.      |
-| Changing the behaviour of a builtin function or type                       | LV                 | This changes the behaviour of existing scripts, so is not backwards compatible.                     |
-| Changing the interface between the ledger and the interpreter              | LV                 | The ledger must provide scripts with exactly the right interface. New interface means new language. |
-| Improving model performance                                                | PP                 | _Must_ strictly follow an improvement in real performance.[^3]                                      |
+| Adding a new Plutus Core language version                                 | HF                 | Backwards-compatible since the new behaviour is guarded by the new LV. |
```

Comment:
Doesn't this depend on what changed in the new LV? Imagine a new LV that removes builtin type `Data` (by the way, do we still need `Data` once we have sums of products?). This would lead to an unavoidable change in the ledger-script interface, necessitating a new LL.

Reply:
Sure, but then you're also including a change to the ledger-script interface. If you do multiple changes obviously you need to biggest kind of release.

Reply:
Fair enough.